mobile-app/cypress/integration/functional/wallet/loans/loans.spec.ts

```diff
@@ -294,222 +346,261 @@ context('Wallet - Loans - Payback Loans', () => {
-    cy.getByTestID('text_vault_id').contains(vaultId)
-    cy.getByTestID('text_resulting_loan_amount').should('have.text', '0.00000000')
-    cy.getByTestID('text_resulting_loan_amount_suffix').should('have.text', 'DUSD')
+  it('should display 100% of loan amount if MAX button is pressed with sufficient DUSD', () => {
+    validateMaxButtonWithSufficientBalance('DUSD')
+  })
+
+  it('should display excess amount details when paying with DUSD', () => {
+    cy.getByTestID('payback_input_text').clear().type('20').blur()
     cy.getByTestID('loan_outstanding_balance').invoke('text').then(text => {
       const outstandingBalance = new BigNumber(text.replace('DUSD', '').trim())
-      const excessAmount = new BigNumber(102).minus(outstandingBalance)
-      const resultingBalance = new BigNumber(110).minus(102).plus(excessAmount)
+      const excessAmount = new BigNumber(20).minus(outstandingBalance)
       cy.getByTestID('text_excess_amount').should('have.text', excessAmount.toFixed(8))
-      // wallet balance - amount to pay + excess amount
-      cy.getByTestID('text_resulting_balance').should('have.text', resultingBalance.toFixed(8))
     })
-    cy.getByTestID('estimated_fee').contains('0.0002')
-    cy.getByTestID('estimated_fee_suffix').should('have.text', 'DFI')
   })
 
-  it('should display tx details in paying DFI w/o excess payment', function () {
-    cy.getByTestID('payment_token_card_DFI').click()
-    cy.getByTestID('payback_input_text').clear().type('50').blur()
-    cy.getByTestID('payback_input_text_error').should('not.exist')
-    cy.getByTestID('text_penalty_fee_warning').should('exist')
-    cy.getByTestID('text_amount_to_pay').should('have.text', '50.50505051')
-    cy.getByTestID('text_amount_to_pay_suffix').should('have.text', 'DUSD')
-    cy.getByTestID('text_amount_to_pay_converted').should('have.text', '0.50505051')
-    cy.getByTestID('text_amount_to_pay_converted_suffix').should('have.text', 'DFI')
-    cy.getByTestID('text_resulting_balance_label').should('have.text', 'Resulting DFI Balance')
-    cy.getByTestID('text_resulting_balance_suffix').should('have.text', 'DFI')
-    /* TODO: Failing e2e - balances page not finishes loading */
-    // cy.getByTestID('bottom_tab_balances').click()
-    // cy.getByTestID('dfi_total_balance_amount').invoke('text').then(text => {
-    //   const dfiBalance = new BigNumber(text)
-    //   cy.getByTestID('bottom_tab_loans').click()
-    //   cy.getByTestID('text_resulting_balance').should('have.text', dfiBalance.minus('0.50505051').toFixed(8))
-    // })
-    cy.getByTestID('text_vault_id').contains(vaultId)
-    cy.getByTestID('loan_outstanding_balance').invoke('text').then(text => {
-      const outstandingBalance = new BigNumber(text.replace('DUSD', '').trim())
-      cy.getByTestID('text_resulting_loan_amount').should('have.text', outstandingBalance.minus('50.50505051').toFixed(8))
+  // it('should be able to payback DUSD loans with DUSD', function () {
```

Comment:
Can add a comment here to indicate why we comment it for now

Reply:
Updated!